File: controller/src/main/java/yeetcd/controller/pipeline/PreviousWork.java
package yeetcd.controller.pipeline;

import yeetcd.protocol.pipeline.PipelineOuterClass;

public record PreviousWork(Work work, String outputPathsMount, String stdOutEnvVar) {

    public static PreviousWork fromProtobuf(PipelineOuterClass.PreviousWork previousWork) {
        return new PreviousWork(
                Work.fromProtobuf(previousWork.getWork()),
                previousWork.getOutputPathsMount(),
                previousWork.getStdOutEnvVar()
        );
    }
}

File: core/internal/core/pipeline/previous_work.go
package pipeline

// PreviousWork represents a dependency on previous work
type PreviousWork struct {
	Work             Work
	OutputPathsMount string
	StdOutEnvVar     string
}

// FromProtobuf converts protobuf PreviousWork to Go struct
func PreviousWorkFromProtobuf(protoPreviousWork interface{}) (*PreviousWork, error) {
	return nil, nil
}
